cpg_workflows/metamist.py

```diff
@@ -435,12 +436,17 @@ def get_analyses_by_sgid(
         """
         metamist_proj = self.get_metamist_proj(dataset)
 
+        # Dynamically retrieve the correct stage_name field based on analysis_type
+        stage_name_key = f"{analysis_type.value.lower()}_stage_name"
+        stage_name = meta.get(stage_name_key) if meta else None
+
         analyses = query(
             GET_ANALYSES_QUERY,
             variables={
                 'metamist_proj': metamist_proj,
                 'analysis_type': analysis_type.value,
                 'analysis_status': analysis_status.name,
+                'stage_name': stage_name,
```

Comment:
I've tested this out, and when you pass `None` as a variable to the meta filter it has the same effect as if the stage name filter were not present - which I think is the intention, ie. if the stage name is not in the config then we shouldn't filter by stage name. But - this feels a bit flaky as generally I'd expect passing `None` as the value for an `eq` filter would return items where that field's value equals `None`, it almost feels like a bug in metamist. 

I think it would be more robust to not include the stage name filter in the query if the stage name value is `None`.  Would need to change `GET_ANALYSES_QUERY` to be a function that accepts a boolean like `include_stage_filter` or something to do this though.

Reply:
Good point! Fortunately no longer required given the refactor from other feedback above :)